pkg/plugin/aggregation/update.go

```diff
@@ -157,3 +161,24 @@ func GetPatch(annotation string) map[string]interface{} {
 		},
 	}
 }
+
+// Set the sonobuoy master pod name
+// The sonobuoy master pod is identified by the label 'run=sonobuoy-master`
+// This function retrieves the pod name based on that label and sets it
+// provided there is only one pod returned with that label
+func SetStatusPodName(client kubernetes.Interface, namespace string) {
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "run=sonobuoy-master",
+	}
+	podList, err := client.CoreV1().Pods(namespace).List(listOptions)
+	if err != nil {
+		logrus.Errorf("Error listing pods with label 'run:sonobuoy-master': %s", err)
+	} else {
+		if len(podList.Items) != 1 {
+			logrus.Error("More than one pod with label 'run:sonobuoy-master'")
```

Comment:
I'd modify this to
 - use the constant value for the label
 - inform the user as to the consequence; in this situation, It seems like we should choose a reasonable default which I think should be the first item in the list (rather than leaving the default as-is, which may not be one of the choices).
 - be sure to handle the `err==nil && len(items)==0` case. I'm not certain if k/k returns empty list w/o error or not but I'd just be defensive.

Also, I'd adjust the flow control a bit to simplify the indenting here:
```
// get list
if err != nil {
  log error and return
}
switch{
case len(items) == 0 :
  error and state we'll use the default value (and show it to user)
case len(items) > 1:
  log situation as a warning (I wouldn't use error) and choose the first one in the list (and let user know the choice)
default:
  choose first (and only item) without any logging
}
```

Reply:
Thanks for the detailed suggestion @johnSchnake.  I've added a commit that addresses this.